multiapi-engine/src/main/java/com/sngular/api/generator/plugin/asyncapi/util/MapperContentUtil.java

```diff
@@ -222,32 +238,96 @@ private static SchemaFieldObject processFieldObjectList(final String propertyNam
         if (items.has("$ref")) {
           modelToBuildList.add(MapperUtil.getRefClass(items));
         }
+        final Iterator<Map.Entry<String, JsonNode>> iterator = schema.fields();
+        Entry<String, JsonNode> current = null;
         fieldObject =
-          SchemaFieldObject
-            .builder()
-            .baseName(name)
-            .dataType(arrayType)
-            .dataTypeSimple(type)
-            .importClass(getImportClass(arrayType))
-            .build();
+            SchemaFieldObject
+                .builder()
+                .baseName(name)
+                .dataType(arrayType)
+                .dataTypeSimple(type)
+                .importClass(getImportClass(arrayType))
+                .build();
+        while (iterator.hasNext()) {
+          current = iterator.next();
+          switch (current.getKey()) {
+            case "maxItems":
+              fieldObject.setMaxItems(current.getValue().intValue());
+              break;
+            case "minItems":
+              fieldObject.setMinItems(current.getValue().intValue());
+              break;
+            case "uniqueItems":
+              fieldObject.setUniqueItems(current.getValue().booleanValue());
+              break;
+            default:
+              break;
+          }
+        }
+        fieldObject.setRequired(required);
       } else if (schema.has("enum")) {
         fieldObject = processEnumField(name, required, schema, prefix, suffix);
       } else {
+        final Iterator<Map.Entry<String, JsonNode>> iterator = schema.fields();
+        Entry<String, JsonNode> current = null;
         fieldObject = SchemaFieldObject
-          .builder().baseName(name).dataType(MapperUtil.getSimpleType(schema, prefix, suffix)).dataTypeSimple(MapperUtil.getSimpleType(schema, prefix, suffix)).build();
+                          .builder().baseName(name).dataType(MapperUtil.getSimpleType(schema, prefix, suffix)).dataTypeSimple(MapperUtil.getSimpleType(schema, prefix, suffix))
+                          .build();
+        while (iterator.hasNext()) {
+          current = iterator.next();
+          switch (current.getKey()) {
+            case "minimum":
+              fieldObject.setMinimum(current.getValue().decimalValue());
```

Comment:
This function is incredibly long, could we move this into its own method?

Reply:
Done